### PR TITLE
Fix restoring files ownership as non-root user

### LINKED ISSUE #655: error for restore: lchown operation not permitted
Restore operation from a root user backup does execute and all files appear to be restored however this error occurs if you don't run the restore operation as a root user.

## Output of `restic version`
```
[root@alarmpi ~]# restic version
restic 0.3.0 (v0.3.0-6-g6f72164)
compiled at 2016-10-05 09:06:15 with go1.7.1 on linux/arm
```

```

[atomi@workstation ~]# restic version
restic 0.3.0 (v0.3.0-19-g6b88d3b)
compiled at 2016-10-31 18:33:33 with go1.7.3 on linux/amd64
```

## Expected behavior
restore should function without error stack trace

## Actual behavior
```
[atomi@workstation ~]# restic -r test snapshots
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
67d6c948  2016-11-01 15:33:13  alarmpi                 /root/backup

[atomi@workstation ~]# restic -r test restore 67d6c948 --target backup
restoring <Snapshot 67d6c948 of [/root/backup] at 2016-11-01 15:33:13.667132217 -0700 PDT by root@alarmpi> to backup
error for backup/backup: lchown backup/backup: operation not permitted
Lchown
restic.Node.restoreMetadata
        /tmp/restic-build-431614805/src/restic/node.go:147
restic.(*Node).CreateAt
        /tmp/restic-build-431614805/src/restic/node.go:134
restic.(*Restorer).restoreNodeTo
        /tmp/restic-build-431614805/src/restic/restorer.go:102
restic.(*Restorer).restoreTo
        /tmp/restic-build-431614805/src/restic/restorer.go:53
restic.(*Restorer).RestoreTo
        /tmp/restic-build-431614805/src/restic/restorer.go:122
main.runRestore
        /tmp/restic-build-431614805/src/cmds/restic/cmd_restore.go:135
main.glob..func13
        /tmp/restic-build-431614805/src/cmds/restic/cmd_restore.go:23
github.com/spf13/cobra.(*Command).execute
        /tmp/restic-build-431614805/src/github.com/spf13/cobra/command.go:599
github.com/spf13/cobra.(*Command).ExecuteC
        /tmp/restic-build-431614805/src/github.com/spf13/cobra/command.go:689
github.com/spf13/cobra.(*Command).Execute
        /tmp/restic-build-431614805/src/github.com/spf13/cobra/command.go:648
main.main
        /tmp/restic-build-431614805/src/cmds/restic/main.go:41
runtime.main
        /usr/local/go/src/runtime/proc.go:183
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:2086
```

## Steps to reproduce the behavior

1. restic backup path using root user and owned by root
2. restore using non-root user

This may be related to #244

### PR DESCRIPTION
**Pull request by middelink**
Like tar, it should be possible to restore files and skip the setting ownership of files for non-root users.

Fixes #655

### CODE REVIEW

File: src/restic/node.go [hunk 1]
```diff
@@ -160,32 +160,31 @@ func (node *Node) CreateAt(path string, repo Repository, idx *HardlinkIndex) err
...
 
-	err = node.restoreExtendedAttributes(path)
-	if err != nil {
+	if err = node.restoreExtendedAttributes(path); err != nil {
 		debug.Log("error restoring extended attributes for %v: %v", path, err)
 		return err
 	}
 
+	// Do ownership last and only if you are acting as root.
+	if os.Getuid() == 0 {
```
Comment (fd0) [inline src/restic/node.go]: That's not a good test, `EUID == 0` is sometimes sufficient, or having `CAP_CHOWN` available. I'd just try it last, and return the error (I have no better idea for it).

**Review by fd0 — CHANGES_REQUESTED** [review]

File: src/restic/node.go [hunk 2]
```diff
@@ -160,32 +160,29 @@ func (node *Node) CreateAt(path string, repo Repository, idx *HardlinkIndex) err
...
 
-	err = node.restoreExtendedAttributes(path)
-	if err != nil {
+	if err = node.restoreExtendedAttributes(path); err != nil {
 		debug.Log("error restoring extended attributes for %v: %v", path, err)
 		return err
 	}
 
+	// Change ownership last, as we lose permissions on the file.
+	if err = lchown(path, int(node.UID), int(node.GID)); err != nil {
```
Comment (fd0) [inline src/restic/node.go]: For suid files (as in the test that fails), the suid bits are reset when the file is chowned. And the ctime is also changed. So I'd suggest moving `lchown()` to the beginning of the function again, but only return the error at the end.

**Review by fd0 — CHANGES_REQUESTED** [review]

File: src/restic/node.go (hunk 2 above)
Comment (middelink) [inline src/restic/node.go]: Yeah, live suxs, i'll look into recoding the recoded code... I got sidetracked a bit by transformation functions.

Comment (fd0) [inline src/restic/node.go]: You don't need to apologize, thanks for your work!

**Comment by middelink** [comment]
Done, this works locally to my personal satisfaction :)

**Review by fd0 — APPROVED** [review]

**Comment by fd0** [comment]
Thanks!

**Pull request merged**